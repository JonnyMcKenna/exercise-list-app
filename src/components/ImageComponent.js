import "../App.css";
import * as constants from "../constants";

const ImageComponent = (props) => {
  const {
    male = { image: "" },
    female = { image: "" },
    gender,
    maleClassName,
    femaleClassName,
  } = props;

  return (
    <div>
      {gender === constants.MALE && (
        <img className={maleClassName} src={male.image} alt="Male" />
      )}

      {gender === constants.FEMALE && (
        <img className={femaleClassName} src={female.image} alt="Female" />
      )}
    </div>
  );
};

export default ImageComponent;
