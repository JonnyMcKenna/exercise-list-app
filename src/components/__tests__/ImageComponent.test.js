import React from 'react'
import ImageComponent from '../ImageComponent'
import {render } from '@testing-library/react';   

describe('ImageComponent', () => {
 
 it('should render male image when gender is male', () => {
   const appElement = render(<ImageComponent gender='male' />).baseElement;
   expect(appElement).toMatchSnapshot()
   });

   it('should render female image when gender is female', () => {
    const appElement = render(<ImageComponent gender='female' />).baseElement;
    expect(appElement).toMatchSnapshot()
    });

  });