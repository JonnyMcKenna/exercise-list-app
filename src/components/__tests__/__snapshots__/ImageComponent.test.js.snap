// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent should render female image when gender is female 1`] = `
<body>
  <div>
    <div>
      <img
        alt="Female"
        src=""
      />
    </div>
  </div>
</body>
`;

exports[`ImageComponent should render male image when gender is male 1`] = `
<body>
  <div>
    <div>
      <img
        alt="Male"
        src=""
      />
    </div>
  </div>
</body>
`;
